statsd/statsd.go

```diff
@@ -133,10 +135,43 @@ func (c *Client) append(cmd string) error {
 	return nil
 }
 
+func joinMaxSize(a []string, sep string, maxSize int) []string {
+	if len(a) == 0 {
+		return []string{""}
+	}
+	if len(a) == 1 {
+		return a
+	}
+	n := len(sep) * (len(a) - 1)
+	for i := 0; i < len(a); i++ {
+		n += len(a[i])
+	}
+	splitSize := (n / maxSize) + 1
+	lastIndex := 0
+	container := make([]string, splitSize)
+	n = -1
+	sepLen := len(sep)
+	for i := 0; i < len(a); i++ {
+		if n+len(a[i])+sepLen > maxSize {
+			container = append(container, strings.Join(a[lastIndex:i], sep))
```

Comment:
This only adds things to `container` when we hit maxSize, which means it will miss all the trailing entries.